modules/library/src/main/scala/zio/elasticsearch/executor/HttpExecutor.scala

```diff
@@ -337,16 +337,20 @@ private[elasticsearch] final class HttpExecutor private (esConfig: ElasticConfig
     }
 
   private def executeSearch(r: Search): Task[SearchResult] = {
-    val body = r.sortBy match {
-      case sorts if sorts.nonEmpty =>
-        Obj(
-          List(
-            "query" -> r.query.paramsToJson(fieldPath = None),
-            "sort"  -> Arr(sorts.toList.map(_.paramsToJson): _*)
-          ): _*
-        )
-      case _ =>
-        r.query.toJson
+    val sort =
+      if (r.sortBy.isEmpty) {
+        None
+      } else {
+        Some(Obj("sort" -> Arr(r.sortBy.toList.map(_.paramsToJson): _*)))
+      }
+
+    val highlights = r.highlights.map(_.toJson)
+
+    val body = (sort, highlights) match {
+      case (Some(s), Some(h)) => r.query.toJson merge s merge h
+      case (Some(s), _)       => r.query.toJson merge s
+      case (_, Some(h))       => r.query.toJson merge h
+      case _                  => r.query.toJson
```

Comment:
I would use explicit `None` when pattern matching and use `_` only when it's for multiple values, so that cases could be easily reordered without changing the meaning.